src/gui/dataModelTableInterface.cpp

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2024 Team Dissolve and contributors
+
+#include "gui/dataModelTableInterface.h"
+#include "procedure/nodes/node.h"
+
+DataModelTableInterface::DataModelTableInterface(DataModel::Base &dataModel) : dataModel_(dataModel)
+{
+    dataModel_.setMutationSignalFunction([this](DataModel::Base::MutationSignal signal, int startIndex, int endIndex)
+                                         { dataMutated(signal, startIndex, endIndex); });
+}
+
+/*
+ * QAbstractTableModel Overrides
+ */
+
+// Return row count
+int DataModelTableInterface::rowCount(const QModelIndex &parent) const { return dataModel_.nDataItems(); }
+
+// Return column count
+int DataModelTableInterface::columnCount(const QModelIndex &parent) const { return dataModel_.nProperties(); }
+
+// Return flags for the specified model index
+Qt::ItemFlags DataModelTableInterface::flags(const QModelIndex &index) const
+{
+    auto &propertyFlags = dataModel_.propertyFlags(index.column());
+    Qt::ItemFlags flags = Qt::ItemIsSelectable;
+    if (!propertyFlags.isSet(DataModel::ItemProperty::ReadOnly))
+        flags.setFlag(Qt::ItemIsEditable);
+    if (!propertyFlags.isSet(DataModel::ItemProperty::Disabled))
+        flags.setFlag(Qt::ItemIsEnabled);
+    return flags;
+}
+
+// Return header data for the specified section, orientation, and role
+QVariant DataModelTableInterface::headerData(int section, Qt::Orientation orientation, int role) const
+{
+    if (role != Qt::DisplayRole || orientation != Qt::Horizontal)
+        return {};
+
+    return QString::fromStdString(dataModel_.propertyName(section));
+}
+
+// Return data for the index and role specified
+QVariant DataModelTableInterface::data(const QModelIndex &index, int role) const
+{
+    if (role != Qt::DisplayRole && role != Qt::EditRole)
+        return {};
+
+    // Get the specified data property
+    auto property = dataModel_.getProperty(index.row(), index.column());
+
+    // Construct a QVariant from the contents of our std::variant
+    return std::visit(
+        DataModel::PropertyVisitor{[](int arg) { return QVariant(arg); }, [](double arg) { return QVariant(arg); },
+                                   [](std::string_view arg) { return QVariant(QString::fromStdString(std::string(arg))); },
+                                   [](std::string &arg) { return QVariant(QString::fromStdString(arg)); }},
+        property);
```

Comment:
```suggestion
    return std::visit(
        [](auto arg) {
            using T = std::decay_t<decltype(arg)>;
            if constexpr (std::is_same_v<T, std::string_view>)
                return QVariant(QString::fromStdString(std::string(arg))); 
            else if constexpr (std::is_same_v<T, std::string>)
                return QVariant(QString::fromStdString(arg)); 
            else
                return QVariant(arg);
        },
        property);
```

I think that we should be able to do this and get rid of the `PropertyVisitor` class, since this is the only place where it is used.

Reply:
And I am very happy to do so since this was voodoo code copy/pasted from cppreference.com.  I cannot pretend that I fully understood the mechanics of it anyway!